setup.py

```diff
@@ -33,7 +34,7 @@ def package_files(directory):
         'databricks-cli>=0.8.0',
         'requests>=2.17.3',
         'six>=1.10.0',
-        'gunicorn',
+        'waitress' if sys.platform == 'win32' else 'gunicorn',
```

Comment:
Why `waitress` only for windows system? Why not let the user choose one or the other for linux systems? I'm not a big fan of gunicorn and I'd like to replace it with an other server. Since it is embedded in mlflow, that would be neat if user could specify his own...